internal/engine/docker_compose_log_manager.go

```diff
@@ -142,7 +142,7 @@ type DockerComposeLogActionWriter struct {
 func (w DockerComposeLogActionWriter) Write(p []byte) (n int, err error) {
 	w.store.Dispatch(DockerComposeLogAction{
 		ManifestName: w.manifestName,
-		Log:          append([]byte{}, p...),
```

Comment:
This seems unnecessary to me?

Reply:
from the interface docs:
https://golang.org/pkg/io/#Writer
"Implementations must not retain p"

slices are mutable, so you don't have any real guarantee that this byte slice will be the same later, when the action is processed

Reply:
aha! Thank you, I learned something new. :)